Assets/Scripts/Utilities/Settings.cs

```diff
@@ -37,31 +34,34 @@ public static class Settings
 
     private static readonly string DefaultSettingsFilePath = System.IO.Path.Combine(
         Application.streamingAssetsPath, System.IO.Path.Combine("Settings", "Settings.xml"));
-    
+
     private static Dictionary<string, string> settingsDict;
 
     private static string userSettingsFilePath = System.IO.Path.Combine(
         Application.persistentDataPath, "Settings.xml");
-    
-    public static string GetSetting(string key, string defaultValue)
+
+    static Settings()
     {
-        // If we haven't already loaded our settings do it now.
-        if (settingsDict == null) 
-        {
-            LoadSettings();
-        }
+        LoadSettings();
+    }
 
-        string s = GetSetting(key);
-        if (s == null)
+    public static string GetSettingWithOverwrite(string key, string defaultValue)
+    {
+        if (settingsDict == null)
         {
-            settingsDict.Add(key, defaultValue);
-            SaveSettings();
+            Debug.ULogErrorChannel("Settings", "Settings Dictionary was not loaded!");
             return defaultValue;
         }
-        else
+
+        string setting = GetSetting(key, defaultValue);
+        if (setting != null && setting != defaultValue)
```

Comment:
As it is, it causes the localization not to load properly for me which generates a whole string of errors.